config/initializers/elasticsearch.rb

```diff
@@ -0,0 +1,11 @@
+require 'typhoeus/adapters/faraday'
```

Comment:
Hmm, good question! This allows the Elasticsearch API to use keep-alive connections and non-blocking requests via [Typhoeus](https://github.com/typhoeus/typhoeus) configured with Faraday as its adapter, as recommended in the [elasticsearch-transport doc](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-transport). The transport lib is included by the `elasticsearch-model` gem, which is the official Ruby client used by Estella. From the doc:

> For optimal performance, use a HTTP library which supports persistent ("keep-alive") connections, such as Typhoeus. Just require the library (require 'typhoeus'; require 'typhoeus/adapters/faraday') in your code, and it will be automatically used; currently these libraries will be automatically detected and used: Patron, HTTPClient and Net::HTTP::Persistent.

We can check the transport, which shows that Typhoeus is being used as the connection handler, so it's not just the Faraday adapter that is being detected and used:

```ruby
Elasticsearch::Model.client.transport
=> #<Elasticsearch::Transport::Transport::HTTP::Faraday:0x007f8ff3c29270
Elasticsearch::Model.client.transport.connections.first
=> #<Elasticsearch::Transport::Transport::Connections::Connection:0x007f8ff3c3b5d8
 @connection=
  #<Faraday::Connection:0x007f8ff3c28fa0
   @builder=#<Faraday::RackBuilder:0x007f8ff3c28c08 @handlers=[Faraday::Adapter::Typhoeus]>,
  ...
```

Typhoeus gives us non-blocking requests, so I think it's probably worth using for that fact alone.